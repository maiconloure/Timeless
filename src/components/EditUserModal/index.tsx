import { Input, Button } from 'capstone-project';
import React from 'react';
import { icons } from 'utils';

import { EditUserModalProps } from '../ComponentsInterface';
import * as St from '../EditModalStyles';
import ModalBackground from '../ModalBackground';

const EditUserModal = ({
  showEditUser,
  showMobileMenu,
  setShowEditUser,
  userName,
  userAbout,
  userImage,
  submitNotification,
  handleUserName,
  handleUserAbout,
  handleUserImage,
}: EditUserModalProps) => (
  <>
    <ModalBackground show={showEditUser} />
    <St.CardModal
      icon={icons.closeWindow}
      title="Editar"
      data={[showEditUser, setShowEditUser]}
      styles={{
        size: showMobileMenu ? 'normal' : 'large',
        fontSize: showMobileMenu ? 'large' : 'largest',
        bgColorPrimary: '#cdd9e2',
        colorPrimary: '#014d82',
      }}>
      <div>
        <St.Form>
          {!showMobileMenu && <h3>Seu Nome</h3>}

          <Input
            type="text"
            placeholder={userName}
            width="220px"
            fontSize="2rem"
            height="40px"
            onTextChange={handleUserName}
          />
          {!showMobileMenu && <h3>Sobre VocÃª</h3>}

          <Input
            type="text"
            placeholder={userAbout}
            width="220px"
            fontSize="2rem"
            height="40px"
            onTextChange={handleUserAbout}
          />
          {!showMobileMenu && <h3>Link da Imagem</h3>}

          <Input
            type="text"
            placeholder={userImage}
            width="220px"
            fontSize="2rem"
            height="40px"
            onTextChange={handleUserImage}
          />

          <Button fontSize="2.6rem" height="44px" weight={600} onClick={submitNotification}>
            Modificar
          </Button>
        </St.Form>
      </div>
    </St.CardModal>
  </>
);

export default EditUserModal;
