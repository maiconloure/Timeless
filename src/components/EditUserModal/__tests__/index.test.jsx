import EditUserModal from '../index';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Snapshot', () => {
  it('renders', () => {
    const tree = renderer
      .create(
        <EditUserModal
          showEditUser={true}
          setShowEditUser={jest.fn()}
          userName={'Test'}
          userAbout={'Testing test.'}
          userImage={'test'}
          submitNotification={jest.fn()}
          handleUserName={jest.fn()}
          handleUserAbout={jest.fn()}
          handleUserImage={jest.fn()}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
