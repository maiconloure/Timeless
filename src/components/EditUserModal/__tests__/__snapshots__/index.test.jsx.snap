// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
Array [
  <div
    className="sc-efHXLn dAUWVC"
  />,
  <div
    className="sc-fKgIGh kEPpFa"
  >
    <div
      className="sc-crzoUp kYmTKt"
      onClick={[Function]}
    >
      <img
        alt="close icon"
      />
    </div>
    <h1
      className="sc-bCwgka iFtFye"
    >
      Editar
    </h1>
    <div
      className="sc-iwaifL sc-cxNIbT fFSxYx hcDzte"
    >
      <div>
        <div
          className="sc-cOigif hsgXtx"
        >
          <h3>
            Seu Nome
          </h3>
          <div
            className="sc-dlnjPT"
          >
            <input
              className="sc-hKFyIo evVGth"
              color="#014D82"
              fontSize="2rem"
              height="40px"
              name="fname"
              onChange={[Function]}
              placeholder="Test"
              type="text"
              width="220px"
            />
          </div>
          <h3>
            Sobre VocÃª
          </h3>
          <div
            className="sc-dlnjPT"
          >
            <input
              className="sc-hKFyIo evVGth"
              color="#014D82"
              fontSize="2rem"
              height="40px"
              name="fname"
              onChange={[Function]}
              placeholder="Testing test."
              type="text"
              width="220px"
            />
          </div>
          <h3>
            Link da Imagem
          </h3>
          <div
            className="sc-dlnjPT"
          >
            <input
              className="sc-hKFyIo evVGth"
              color="#014D82"
              fontSize="2rem"
              height="40px"
              name="fname"
              onChange={[Function]}
              placeholder="test"
              type="text"
              width="220px"
            />
          </div>
          <button
            className="sc-bdnylx jmFhMj"
            color="#014D82"
            fontSize="2.6rem"
            height="44px"
            onClick={[MockFunction]}
            radius="6px"
            width="170px"
          >
            Modificar
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
